test-requirements.txt

```diff
@@ -3,7 +3,7 @@ coverage[toml]
 flake8-docstrings
 flake8-builtins
 isort
-juju==2.9.42.2
+juju==3.1.2.0
```

Comment:
Why are we pinning the version?

Reply:
Becaues pythonlibjuju doesn't guarantee compatibility. We pin the version of Juju itself in the integration tests. If we don't pin the `pyhtonlib-juju` version, it's very possible that the 2 won't be compatible very soon.